### ISSUE TITLE
react-native-css-interop issue with react-native-safe-area-context when building for web /w expo-router

### ISSUE THREAD

**Issue by arbel03** [nativewind/nativewind#1069]
Edit: steps to reproduce https://github.com/nativewind/nativewind/issues/1069#issuecomment-2348826477

**Describe the bug**
When I build I get the following:

The error: 
```bash
Web Bundling failed 9188ms node_modules/expo-router/entry.js (1284 modules)

Metro error: Unable to resolve module react-native/Libraries/Utilities/Platform from /Users/arbel03/Developer/monorepo/OnlineTestMakerMobile/node_modules/react-native-css-interop/dist/runtime/third-party-libs/react-native-safe-area-context.js: react-native/Libraries/Utilities/Platform could not be found within the project or in these directories:
  node_modules/react-native-css-interop/node_modules
  node_modules
   8 | exports.SafeAreaProviderShim = SafeAreaProviderShim;
   9 | const react_1 = require("react");
> 10 | const Platform_1 = __importDefault(require("react-native/Libraries/Utilities/Platform"));
     |                                             ^
  11 | const styles_1 = require("../native/styles");
  12 | let SafeAreaLibrary = undefined;
  13 | try {

  1 | "use strict";
  2 | var __importDefault = (this && this.__importDefault) || function (mod) {
  3 |     return (mod && mod.__esModule) ? mod : { "default": mod };

Call Stack
  <unknown> (node_modules/react-native-css-interop/dist/runtime/third-party-libs/react-native-safe-area-context.js:0)
λ Bundling failed 9276ms node_modules/expo-router/node/render.js (1302 modules)
Web Bundling failed 1407ms node_modules/expo-router/_error.js (333 modules)
Unable to resolve "react-native/Libraries/Utilities/Platform" from "node_modules/react-native-css-interop/dist/runtime/third-party-libs/react-native-safe-area-context.js"
```

Packages:
```json
  "dependencies": {
    "@babel/plugin-transform-react-jsx": "^7.25.2",
    "@expo/metro-runtime": "~3.2.3",
    "@expo/webpack-config": "~19.0.1",
    "@invertase/react-native-apple-authentication": "^2.2.2",
    "@notifee/react-native": "^7.8.2",
    "@react-native-async-storage/async-storage": "1.23.1",
    "@react-native-firebase/analytics": "^18.7.3",
    "@react-native-firebase/app": "^20.4.0",
    "@react-native/metro-config": "^0.75.0",
    "@react-navigation/bottom-tabs": "^6.6.1",
    "@react-navigation/elements": "^1.3.31",
    "@react-navigation/native": "^6.1.18",
    "@react-navigation/native-stack": "^6.11.0",
    "@reduxjs/toolkit": "^1.9.5",
    "crypto-browserify": "^3.12.0",
    "crypto-js": "^4.2.0",
    "expo": "latest",
    "expo-build-properties": "~0.12.5",
    "expo-constants": "~16.0.2",
    "expo-crypto": "~13.0.2",
    "expo-firebase-analytics": "^8.0.0",
    "expo-font": "~12.0.10",
    "expo-linking": "~6.3.1",
    "expo-router": "~3.5.23",
    "expo-status-bar": "~1.12.1",
    "moment": "^2.29.4",
    "nativewind": "4.1.8",
    "online-test-maker-api": "file:../OnlineTestMakerApi",
    "postcss": "^8.4.45",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-native": "0.74.5",
    "react-native-css-interop": "0.1.7",
    "react-native-device-info": "^10.12.0",
    "react-native-draggable-flatlist": "^4.0.1",
    "react-native-draglist": "^3.5.1",
    "react-native-element-dropdown": "^2.10.2",
    "react-native-gesture-handler": "~2.16.1",
    "react-native-glassfy-module": "^1.6.0",
    "react-native-glassfy-ui-module": "file:../react-native-glassfy-ui-module",
    "react-native-in-app-review": "^4.3.3",
    "react-native-keyboard-aware-scroll-view": "^0.9.5",
    "react-native-reanimated": "~3.10.1",
    "react-native-safe-area-context": "4.10.5",
    "react-native-screens": "3.31.1",
    "react-native-sectioned-multi-select": "^0.10.0",
    "react-native-toast-message": "^2.2.0",
    "react-native-vector-icons": "^10.1.0",
    "react-native-web": "~0.19.10",
    "react-redux": "^8.0.7",
    "redux-persist": "^6.0.0"
  },
  "devDependencies": {
    "@babel/core": "^7.25.2",
    "@babel/preset-env": "^7.20.0",
    "@babel/runtime": "^7.24.0",
    "@react-native/babel-preset": "^0.75.2",
    "@react-native/eslint-config": "0.75.2",
    "@react-native/metro-config": "^0.75.2",
    "@react-native/typescript-config": "0.75.2",
    "@smartface/image-processor": "^1.2.6",
    "@tsconfig/react-native": "^3.0.0",
    "@types/react": "~18.2.79",
    "@types/react-native-vector-icons": "^6.4.18",
    "@types/react-test-renderer": "^18.0.0",
    "autoprefixer": "^10.4.20",
    "babel-jest": "^29.6.3",
    "eslint": "^8.19.0",
    "jest": "^29.2.1",
    "metro-react-native-babel-preset": "0.76.7",
    "prettier": "2.8.8",
    "react-test-renderer": "^18.3.1",
    "tailwindcss": "3.3.2",
    "typescript": "~5.3.3"
  },
```

I also have this in package.json:
```json
  "expo": {
    "install": {
      "exclude": [
        "react-native-safe-area-context"
      ]
    }
  },
```

**Additional context**
I am trying to build for web but from my understanding, there is an interop issue.
I added the react-native-css-interop myself because there was another issue when npm installed it as a dependency of nativewind.

Before I create a reproduction repo, does anyone have a quick fix or a set of commands I can run to try to refresh my
packages? I added the expo:install:exclude in my package.json after installing everything.
Thanks

**Comment by arbel03** [comment]
Steps to reproduce:
Maybe it is a local problem with my setup but the template repo:
  - `npx create-expo-stack@latest --nativewind --expo-router`
Creates a faulty repo which reproduces this error out of the box.
`npm run web`

**Comment by aldebout** [comment]
Maybe related: going from 4.1.7 to 4.1.8, bundling time is higher on expo ios development build and then the splash screen never goes away. Nothing in the logs.

**Comment by cmaycumber** [comment]
I can confirm got this error after updating to 4.1.8

**Comment by BoavistaLudwig** [comment]
I have the same issues since today.

**Comment by MJRT** [comment]
same issue, It's alright after switched to 4.1.7.

**Comment by marklawlor** [comment]
Fixed in `v4.1.9`
